File: 661.image-smoother.cpp
/*
 * @lc app=leetcode id=661 lang=cpp
 *
 * [661] Image Smoother
 */

// @lc code=start
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>>& img) {
        int m = img.size();
        int n = img[0].size();
        vector<vector<int>> result(m, vector<int>(n, 0));

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int avg = 0;
                int counter = 0;

                for (int k = -1; k <= 1; k++) {
                    for (int l = -1; l <= 1; l++) {
                        if (0 <= i + k && i + k < m && 0 <= j + l && j + l < n) {
                            avg += img[i + k][j + l];
                            counter++;
                        }
                    }
                }

                result[i][j] = avg / counter;
            }
        }

        return result;
    }
};

// @lc code=end

File: 661.image-smoother.py
#
# @lc app=leetcode id=661 lang=python3
#
# [661] Image Smoother
#

# @lc code=start
from typing import List


class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        m, n = len(img), len(img[0])
        result = [[0 for j in range(n)] for i in range(m)]
        for i in range(len(img)):
            for j in range(len(img[0])):
                avg = 0
                counter = 0
                for k in range(-1,2,1):
                    for l in range(-1,2,1):
                        if 0 <= i + k < len(img) and 0 <= j + l < len(img[0]):
                            avg += img[i+k][j+l]
                            counter += 1
                result[i][j] = avg // counter
        return result
# @lc code=end

